app/models/user.rb

```diff
@@ -141,4 +150,23 @@ def is_editor?
     %w( editor agent admin ).include?(self.role)
   end
 
+  def self.bulk_invite(emails, message)
+    #below line merge comma saperated emails as well as emails saperated by new lines
+    emails = emails.each_line.reject { |l| l =~ /^\s+$/ }.map { |l| l.strip.split(', ') }.flatten
+    
+    emails.each do |email|
+      is_valid_email = email.match('^.+@.+$')
+      if is_valid_email
+        User.invite!({email: email}) do |user|
+          user.message = message
+        end
+      end
+    end
+  end
+  
+  #when using deliver_later attr_accessor :message becomes nil on mailer view
+  def send_devise_notification(notification, *args)
+    devise_mailer.send(notification, self, *args).deliver
+  end
+
```

Comment:
I saw one response about this on SO suggesting to serialize the message into the job, which seems right to me. Any thoughts @shaktik1989?